bytecomp/translcore.ml

```diff
@@ -1160,15 +1167,21 @@ and transl_apply ?(should_be_tailcall=false) ?(inlined = Default_inline)
         let body =
           match build_apply handle ((Lvar id_arg, optional)::args') l with
             Lfunction{kind = Curried; params = ids; body = lam; attr; loc} ->
```

Comment:
I made the propagation explicit. I tried propagating enough information for this not to be always Pgenval, but this seems too complex for the benefit.